doc/instrumentation.rst

```diff
@@ -269,4 +269,468 @@ Available Instruments
+
+.. code-block:: python
+
+   acme.setup(shunt_resistor=[20000, 10000],
+              integration_time_bus=[int_times[2], int_times[3]],
+              integration_time_shunt=[int_times[3], int_times[4]],
+              oversampling_ratio=[ratios[0], ratios[1]])
+
+   for n, p in acme.probes.iteritems():
+      print('{}:'.format(n))
+      print('    T_bus = {} s'.format(p.integration_time_bus))
+      print('    T_shn = {} s'.format(p.integration_time_shunt))
+      print('    N     = {}'.format(p.oversampling_ratio))
+      print('    freq  = {} Hz'.format(p.sample_rate_hz))
+
+   # Output:
+   #
+   #   battery:
+   #       T_bus = 0.000332 s
+   #       T_shn = 0.000588 s
+   #       N     = 1
+   #       freq  = 1087 Hz
+   #   usb:
+   #       T_bus = 0.000588 s
+   #       T_shn = 0.0011 s
+   #       N     = 4
+   #       freq  = 148 Hz
+
+The on-board computation of the shunt current (which in turn is used for computing the power consumption) is done with fixed-point 16-bit arithmetic. Although useful in embedded applications as it comes at no cost for the user, making these computations inside of the INA226 gives a resolution that is unsurprisingly much lower than what a 64-bit FPU is capable of. For this reason and as the code calling devlib most certainly has access to such hardware, we may want to do the power computation on the host as shown bellow. follows (notice that keeping the current and power channels turned off allows us to reduce the required bandwith by 25%):
+
+.. todo:: Fix the units of ``power`` in the following code if we decide to use native order of magnitude for ``data``
+
+.. code-block:: python
+
+   import time
+   import devlib
+
+   acme = devlib.BaylibreAcmeNetworkInstrument(ACME_IP_ADDR, probe_names=['battery', 'usb'])
+
+   int_times = acme.integration_times_available
+   ratios    = acme.oversampling_ratios_available
+
+   shunt_resistor = 0.01 # [Ohms]
+
+   acme.setup(shunt_resistor=20000, # this value won't matter in the following
+              integration_time_bus=int_times[1],
+              integration_time_shunt=int_times[1],
+              oversampling_ratio=ratios[1])
+
+   acme.reset(probes='battery', sites=['timestamp', 'bus', 'shunt'])
+   acme.start()
+   time.sleep(10)
+   acme.stop()
+   data = acme.get_data()['battery']
+   acme.teardown()
+
+   # Compute numpy array containing the power channel with 64-bit FPU accuracy:
+   power = data['shunt_voltage'] * data['bus_voltage'] / shunt_resistor
```

Comment:
Could we imagine having a `high_resolution_power` option for `BaylibreAcmeInstrument` which would allow computing "offline" the power signal or is this too far from the data acquisition to be incorporated into an `Instrument`?